core/src/main/java/org/infinispan/protostream/TagHandler.java

```diff
@@ -13,8 +15,10 @@
 
    /**
     * Start of top level message. Do whatever required init here.
+    *
+    * @param descriptor The {@link GenericDescriptor} for the top level Enum or Message. For primitives it is null.
     */
-   default void onStart() {
+   default void onStart(AnnotatedDescriptor descriptor) {
```

Comment:
I'd prefer the ```descriptor``` arg to be a ```GenericDescriptor```. There's no need to for this one to be a field descriptor too. I know there is one occurrence where this seems to be needed but that is a mistake. I'll comment on it.